core/src/mbc/mbc5.rs

```diff
@@ -0,0 +1,121 @@
+use crate::{
+    cartridge::{RamSize, RomSize},
+    primitives::{Byte, Word},
+};
+use super::Mbc;
+
+/// MBC5.
+///
+/// With this controller, the cartridge can have up to 8MiB of ROM and up to
+/// 128KiB of external RAM.
+pub(crate) struct Mbc5 {
+    rom: Box<[Byte]>,
+    ram: Box<[Byte]>,
+
+    /// A 9 bit number to select the bank mapped to 0x4000 -- 0x8000. Values 0
+    /// to 0x1E0. In MBC5 you can actually select bank 0 here to map bank 0
+    /// twice. Bits 9 to 15 are always 0.
+    rom_bank: u16,
+
+    /// A 4 bit number to select the RAM bank. Values 0 to 0xF.
+    ram_bank: u8,
+
+    /// Whether or not the RAM is enabled.
+    ram_enabled: bool,
+}
+
+
+impl Mbc5 {
+    pub(crate) fn new(data: &[u8], rom_size: RomSize, ram_size: RamSize) -> Self {
+        assert!(
+            rom_size.len() == data.len(),
+            "Length of cartridge doesn't match length specified in ROM size header",
+        );
+
```

Comment:
Why don't you check for `rom_size <= RomSize::???` here?

Reply:
Because MBC5 supports up to 512 banks which is the largest value. However, I added the check now anyway in case we extend the `RomSize` enum at some point.